asterisk/rootfs/etc/services.d/asterisk_mbox/run

```diff
@@ -15,4 +15,4 @@ fi
 bashio::log.info "Starting Asterisk Mailbox"
 
 ## Run Asterisk_mbox
-exec -u asterisk /usr/local/bin/asterisk-mbox-server --cfg /etc/asterisk/asterisk_mbox.ini
+-u asterisk /usr/local/bin/asterisk-mbox-server --cfg /etc/asterisk/asterisk_mbox.ini
```

Comment:
Didn't you mean:

```suggestion
exec /usr/local/bin/asterisk-mbox-server --cfg /etc/asterisk/asterisk_mbox.ini -u asterisk 
```

?

Reply:
No the mbox-server doesn't have a `-u` option. I fixed it with `su`.

Reply:
Got it. But I believe there is a better option. `su` may introduce additional processes to the container. One sec

Reply:
https://github.com/just-containers/s6-overlay#dropping-privileges

Can you try with:

```suggestion
exec s6-setuidgid asterisk /usr/local/bin/asterisk-mbox-server --cfg /etc/asterisk/asterisk_mbox.ini
```

Reply:
Trying now.

Reply:
I get this.

![image](https://user-images.githubusercontent.com/32220029/153276596-d1d49baf-61d4-479c-a7ad-428c7c82490e.png)

Reply:
This is weird. Perhaps S6-Overlay is too old. Let me check here, one sec.

Reply:
Your output says `s6-envuidgid`, but I suggested `s6-setuidgid`. Are you using `s6-setuidgid`?

Reply:
I have `exec s6-setuidgid asterisk /usr/local/bin/asterisk-mbox-server --cfg /etc/asterisk/asterisk_mbox.ini`.

Reply:
See: https://github.com/hassio-addons/addon-base-python/issues/102

Reply:
Maybe use the normal image and include python ourself? Should be possible in the build-base right?

Reply:
Can you try with:

```suggestion
exec setpriv --reuid=asterisk --regid=asterisk --init-groups -- /usr/local/bin/asterisk-mbox-server --cfg /etc/asterisk/asterisk_mbox.ini
```

It should be a drop-in replacement.

Reply:
Refs https://github.com/hassio-addons/addon-base-python/pull/103

Reply:
Get this:
![image](https://user-images.githubusercontent.com/32220029/153304223-2701bf28-8734-4e37-a166-64b29a1c8a99.png)

Also tried with quotes: `exec setpriv --reuid=asterisk --regid=asterisk --init-groups -- '/usr/local/bin/asterisk-mbox-server --cfg /etc/asterisk/asterisk_mbox.ini'`
same result...

Reply:
I don't get it. Did you left `-U asterisk`?

Reply:
Hmm... It's acting weird now. I reinstall the local add-on to test but it doesn't use the new files anymore. Will try a new version so it updates.

Reply:
Maybe you forgot to comment out the `image:` in the config.yaml?

Reply:
Oh! `setpriv` is missing a lot of things there.

Reply:
I think I have a workaround. Let me push a commit in this PR.

Reply:
Done. It should work now. :D

Reply:
Yes, it seem's to be working now. Can't test the mailbox server itself right now because it requires a billing account (it's still free, kinda) which I can't set up at the moment.

Thanks for the help! 😄

Reply:
np :P